src/bulk/common.ts

```diff
@@ -328,34 +329,61 @@ export class BulkWriteResult {
   }
 }
 
+/** @internal */
+export interface WriteConcernErrorData {
+  code: number;
+  errmsg: string;
+  errInfo?: Document;
+}
+
 /**
  * An error representing a failure by the server to apply the requested write concern to the bulk operation.
  * @public
  * @category Error
  */
 export class WriteConcernError {
-  err: MongoServerError;
+  /** @internal */
+  [kServerError]: WriteConcernErrorData;
 
-  constructor(err: MongoServerError) {
-    this.err = err;
+  /** @internal */
+  constructor(error: WriteConcernErrorData) {
+    this[kServerError] = error;
   }
 
   /** Write concern error code. */
   get code(): number | undefined {
-    return this.err.code;
+    return this[kServerError].code;
   }
 
   /** Write concern error message. */
-  get errmsg(): string {
-    return this.err.errmsg;
+  get errmsg(): string | undefined {
+    return this[kServerError].errmsg;
   }
 
-  toJSON(): { code?: number; errmsg: string } {
-    return { code: this.err.code, errmsg: this.err.errmsg };
+  /** Write concern error info. */
+  get errInfo(): Document | undefined {
+    return this[kServerError].errInfo;
+  }
+
+  /**
+   * @internal
+   * @deprecated The `err` prop that contained a MongoServerError has been deprecated.
+   */
+  get err(): WriteConcernErrorData {
+    return this[kServerError];
+  }
+
+  toJSON(): {
+    code: number;
+    errmsg: string;
+    errInfo?: Document;
+  } {
+    // WriteConcernErrorData has been inlined as the return to avoid making it public
```

Comment:
what's the reasoning against make it public?

Reply:
I figured it would be yet another type clouding the namespace relating to an unclearly layered WriteConcernError/WriteError/BulkError space. But it's an 🧅 that can be argued otherwise